tests/integration/features/features_int_test.go

```diff
@@ -127,6 +132,74 @@ var _ = Describe("preconditions", func() {
 
 })
 
+var _ = Describe("feature trackers", func() {
+	Context("ensuring feature trackers indicate status and phase", func() {
+
+		var (
+			dsciSpec *dscv1.DSCInitializationSpec
+		)
+
+		BeforeEach(func() {
+			dsciSpec = newDSCInitializationSpec("default")
+		})
+
+		It("should indicate successful installation in FeatureTracker", func() {
+			// given example CRD installed into env
+			name := "test-resources.openshift.io"
+			verificationFeature, err := feature.CreateFeature("crd-verification").
+				For(dsciSpec).
+				UsingConfig(envTest.Config).
+				PreConditions(feature.EnsureCRDIsInstalled(name)).
+				Load()
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			Expect(verificationFeature.Apply()).To(Succeed())
+
+			// then
+			featureTracker := getFeatureTracker("default-crd-verification")
+			Expect(featureTracker.Status.Conditions).To(HaveCondition(conditionsv1.ConditionAvailable, v1.ConditionTrue, featurev1.FeatureCreated))
+		})
+
+		It("should indicate failure in preconditions", func() {
+			// given
+			name := "non-existing-resource.non-existing-group.io"
+			verificationFeature, err := feature.CreateFeature("crd-verification").
+				For(dsciSpec).
+				UsingConfig(envTest.Config).
+				PreConditions(feature.EnsureCRDIsInstalled(name)).
+				Load()
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			Expect(verificationFeature.Apply()).ToNot(Succeed())
+
+			// then
+			featureTracker := getFeatureTracker("default-crd-verification")
+			Expect(featureTracker.Status.Conditions).To(HaveCondition(conditionsv1.ConditionDegraded, v1.ConditionTrue, featurev1.PreConditions))
+		})
+
+		It("should indicate failure in post-conditions", func() {
+			// given
+			verificationFeature, err := feature.CreateFeature("post-condition-failure").
+				For(dsciSpec).
+				UsingConfig(envTest.Config).
+				PostConditions(func(f *feature.Feature) error {
+					return fmt.Errorf("always fail")
+				}).
+				Load()
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			Expect(verificationFeature.Apply()).ToNot(Succeed())
+
+			// then
+			featureTracker := getFeatureTracker("default-post-condition-failure")
+			Expect(featureTracker.Status.Conditions).To(HaveCondition(conditionsv1.ConditionDegraded, v1.ConditionTrue, featurev1.PostConditions))
```

Comment:
This is a bit cryptic, let's improve it.

Reply:
[9455151](https://github.com/opendatahub-io/opendatahub-operator/pull/801/commits/94551518ce359dce2f402dc0b865accad940aeba)